### PR TITLE
Fix `mark-private-orgs` showing on all organizations

### PR DESCRIPTION
**Pull request by kytwb**
Fixes https://github.com/sindresorhus/refined-github/issues/2830.
Details: https://github.com/sindresorhus/refined-github/issues/2830#issuecomment-591111897.

### CODE REVIEW

**Comment by kytwb** [comment]
I am not sure if this change to the organizations URLs is globally available on GitHub or if we should handle both the old/new URLs?

File: source/features/mark-private-orgs.tsx [hunk 1]
```diff
@@ -22,9 +22,10 @@ async function init(): Promise<false | void> {
 		return false;
 	}
 
-	const publicOrganizations = await getPublicOrganizations(getUsername());
+	const publicOrganizationsNames = await getPublicOrganizationsNames(getUsername());
 	for (const org of orgs) {
-		if (!publicOrganizations.includes(org.pathname.slice(1))) {
+		// https://github.com/sindresorhus/refined-github/issues/2830#issuecomment-591111897
+		if (!publicOrganizationsNames.includes(org.pathname.slice('/organizations/'.length))) {
```
Comment (busches) [inline source/features/mark-private-orgs.tsx]: Looks like a typo here, you're missing a ) after organizations.

Comment (kytwb) [reply source/features/mark-private-orgs.tsx]: It's to slice the value of `'/organizations/'.length`, equivalent to `.slice(15)`. Did that to enhance readability (not to have to guess where the 15 comes from, but now it leads to a syntax error assumption. The irony 😄

Comment (busches) [reply source/features/mark-private-orgs.tsx]: I see, I was thinking it was an array at a quick glance, but it's a string. This approach won't be backwards compatible with GHE, I'm assuming it's rolled out for all of GH, and ideally we remove `/organizations/` in another way that would work even if it exists or not.

Comment (kytwb) [reply source/features/mark-private-orgs.tsx]: Addressed via ff49bdb.

File: source/features/mark-private-orgs.tsx [hunk 2]
```diff
@@ -22,9 +22,10 @@ async function init(): Promise<false | void> {
 		return false;
 	}
 
-	const publicOrganizations = await getPublicOrganizations(getUsername());
+	const publicOrganizationsNames = await getPublicOrganizationsNames(getUsername());
 	for (const org of orgs) {
-		if (!publicOrganizations.includes(org.pathname.slice(1))) {
+		// https://github.com/sindresorhus/refined-github/issues/2830#issuecomment-591111897
+		if (!publicOrganizationsNames.includes(org.pathname.slice(1).replace('organizations/', ''))) {
```
Comment (fregante) [inline source/features/mark-private-orgs.tsx]: Either drop slice or replace. They can do the same thing. `replace(/^[/]organizations[/]/, '')` has clear intent

**Comment by fregante** [comment]
Thank you for the fast PR!

File: source/features/mark-private-orgs.tsx (hunk 2 above)
Comment (kytwb) [reply source/features/mark-private-orgs.tsx]: Good idea! I have incorporated your feedback via 6e6af3a 🙏

**Comment by fregante** [comment]
I'm not seeing these URLs yet; this change would break the feature for me.

**Comment by kytwb** [comment]
@fregante The change has been reverted on my GitHub as well. I have updated the replace regex to work in both cases if it were to be restored.

File: source/features/mark-private-orgs.tsx [hunk 3]
```diff
@@ -22,9 +22,9 @@ async function init(): Promise<false | void> {
 		return false;
 	}
 
-	const publicOrganizations = await getPublicOrganizations(getUsername());
+	const publicOrganizationsNames = await getPublicOrganizationsNames(getUsername());
 	for (const org of orgs) {
-		if (!publicOrganizations.includes(org.pathname.slice(1))) {
+		if (!publicOrganizationsNames.includes(org.pathname.replace(/^\/(organizations\/)?|\/$/, ''))) {
```
Comment (fregante) [inline source/features/mark-private-orgs.tsx]: I think this is equivalent
```suggestion
		if (!publicOrganizationsNames.includes(org.pathname.replace(/^\/(organizations\/)?$/, ''))) {
```

Comment (kytwb) [reply source/features/mark-private-orgs.tsx]: I thought so as well when I first read it (`/` followed by optional `organizations/`), but:

```
> '/organizations/getferdi'.replace(/^\/(organizations\/)?$/, '')
'/organizations/getferdi'
> '/getferdi'.replace(/^\/(organizations\/)?$/, '')
'/getferdi'
```

Then I noticed I mistakenly added the `$`:

```
> '/organizations/getferdi'.replace(/\/(organizations\/)?/, '')
'getferdi'
> '/getferdi'.replace(/\/(organizations\/)?/, '')
'getferdi'
```

Comment (kytwb) [reply source/features/mark-private-orgs.tsx]: Addressed via bf1d679. Thanks for bearing with me! 😄

**Pull request merged**